dev-envs/linux/utils.sh

```diff
@@ -143,9 +143,9 @@ GFOLD_VERSION="4.5.0"
 #     curl "https://github.com/nickgerace/gfold/releases/download/${GFOLD_VERSION}/gfold-linux-gnu-${short_arch}" -Lo /usr/local/bin/gfold
 #     chmod +x /usr/local/bin/gfold
 # else
-#   cargo install gfold@${GFOLD_VERSION}
+#   cargo install --locked gfold@${GFOLD_VERSION}
 # fi
-cargo install gfold@${GFOLD_VERSION}
+cargo install --locked gfold@${GFOLD_VERSION}
```

Comment:
https://gitlab.ddbuild.io/DataDog/datadog-agent-buildimages/-/jobs/825114490#L4616

```
#18 9.863 error: failed to compile `gfold v4.5.0`, intermediate artifacts can be found at `/tmp/cargo-installAnjOp3`.
#18 9.863 To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
#18 9.863 
#18 9.863 Caused by:
#18 9.863   package `zerofrom v0.1.6` cannot be built because it requires rustc 1.81 or newer, while the currently active rustc version is 1.76.0
#18 9.863   Try re-running cargo install with `--locked`
```

Same concept as https://github.com/DataDog/datadog-agent-buildimages/pull/768